### PR TITLE
Feature/add details to join exit pool reponse

### PR DESCRIPTION
**Pull request by Ehsan-saradar**
## What is the purpose of the change

This PR adds more information to response of join/exit pool handlers so the caller of the transaction in ICA receives these information on acknowledgment.

## Brief Changelog

  - *Added `tokenIn sdk.Coins`, `sharesOut sdk.Int` to return of `JoinPoolNoSwap`*
  - *`share_out_amount` and `token_in` fields to `MsgJoinPoolResponse`*
  - *`token_out` field to `MsgExitPoolResponse`*

## Testing and Verifying

This change added tests and can be verified as follows:

  - *Extended integration test for `TestJoinPoolNoSwap`*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

File: x/gamm/keeper/pool_service_test.go [hunk 1]
```diff
@@ -296,7 +298,7 @@ func (suite *KeeperTestSuite) TestJoinPoolNoSwap() {
 				keeper := suite.App.GAMMKeeper
 				// Test the "tokenInMaxs"
 				// In this case, to get the 50 * OneShare amount of share token, the foo, bar token are expected to be provided as 5000 amounts.
-				err := keeper.JoinPoolNoSwap(suite.Ctx, suite.TestAccs[1], poolId, types.OneShare.MulRaw(50), sdk.Coins{
+				_, _, err := keeper.JoinPoolNoSwap(suite.Ctx, suite.TestAccs[1], poolId, types.OneShare.MulRaw(50), sdk.Coins{
```
Comment (mattverse) [inline x/gamm/keeper/pool_service_test.go:301]: Can we add check and test ShareOutAmount and TokenIn returned here is as expected?

File: x/gamm/keeper/pool_service_test.go [hunk 2]
```diff
@@ -493,7 +495,7 @@ func (suite *KeeperTestSuite) TestJoinPoolExitPool_InverseRelationship() {
 		balanceBeforeJoin := suite.App.BankKeeper.GetAllBalances(suite.Ctx, joinPoolAcc)
 		fmt.Println(balanceBeforeJoin.String())
 
-		err = suite.App.GAMMKeeper.JoinPoolNoSwap(suite.Ctx, joinPoolAcc, poolId, tc.joinPoolShareAmt, sdk.Coins{})
+		_, _, err = suite.App.GAMMKeeper.JoinPoolNoSwap(suite.Ctx, joinPoolAcc, poolId, tc.joinPoolShareAmt, sdk.Coins{})
```
Comment (mattverse) [inline x/gamm/keeper/pool_service_test.go:498]: ditto:Can we add check and test ShareOutAmount and TokenIn returned here is as expected?

**Review by mattverse — CHANGES_REQUESTED** [review]
Thanks for this PR! I personally think adding responses to messages are super useful. 

Can we make sure we add new test cases and checks for the new fields to ensure they return expected values?

Also, can you drop the changes for `go.mod` and `go.sum` as well? Otherwise, logic itself 
lgtm!

**Comment by ValarDragon** [comment]
Needs changelog entry as well

**Comment by ValarDragon** [comment]
https://github.com/quasar-finance/osmosis/pull/2

**Review by mattverse — COMMENTED** [review]
Reviewed the changes made in https://github.com/quasar-finance/osmosis/pull/2 Let's merge this once https://github.com/quasar-finance/osmosis/pull/2 gets merged!

**Review by mattverse — APPROVED** [review]

**Pull request merged**